### PR TITLE
Add triangles mesh

### PR DESCRIPTION
**Pull request by tdameros**

### CODE REVIEW

File: src/vectors/vectors3f.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	if (vectors3f->length < vectors3f->size)
+		vectors3f->data[vectors3f->length] = vector3f;
+	else
+	{
+		temp = malloc(sizeof(*vectors3f->data) * (vectors3f->size * 2));
+		if (temp == NULL)
+			return (-1);
+		ft_memcpy(temp, vectors3f->data,
+			sizeof(*vectors3f->data) * vectors3f->size);
```
Comment (V-Fries) [inline src/vectors/vectors3f.c:41]: replace with `ft_memcpy(temp, vectors3f->data, sizeof(*vectors3f->data) * vectors3f->length);` as they is no point in copying the rest

File: src/vectors/mesh_faces.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	if (mesh_faces->length < mesh_faces->size)
+		mesh_faces->data[mesh_faces->length] = mesh_face;
+	else
+	{
+		temp = malloc(sizeof(*mesh_faces->data) * (mesh_faces->size * 2));
+		if (temp == NULL)
+			return (-1);
+		ft_memcpy(temp, mesh_faces->data,
+			sizeof(*mesh_faces->data) * mesh_faces->size);
```
Comment (V-Fries) [inline src/vectors/mesh_faces.c:41]: replace with `ft_memcpy(temp, mesh_faces->data, sizeof(*mesh_faces->data) * mesh_faces->length);` as they is no point in copying the rest

File: src/object/mesh/create.c [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+	mesh_object->type = MESH;
+	mesh_object->material = material;
+	return_code = initialize_mesh_with_obj(&mesh_object->mesh, obj_file);
+	if (return_code < 0)
+		return (-1);
+	else if (return_code == 0)
+		return (0);
+	mesh_object->name = ft_strdup("Mesh");
```
Comment (V-Fries) [inline src/object/mesh/create.c:37]: Maybe use the file name to get a better description?

File: include/object.h [hunk 4]
```diff
@@ -95,6 +98,12 @@ typedef struct s_objects
 	size_t				size;
 }	t_objects;
 
+int	mesh_object_initialize(t_object *mesh_object, const char *obj_file,
+							  t_material material);
+void	mesh_free(t_mesh *mesh);
+t_object	triangle_create(const t_vector3f vertex_a, const t_vector3f vertex_b,
+							const t_vector3f vertex_c, const t_vector3f normal, const t_material material);
```
[the remark this answers is not in the document]
Comment (tdameros) [reply include/object.h:105]: This function is deprecated, I will delete it.

File: src/object/mesh/create.c [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	int	return_code;
+
+	ft_bzero(mesh_object, sizeof(*mesh_object));
+	mesh_object->type = MESH;
+	mesh_object->material = material;
+	return_code = initialize_mesh_with_obj(&mesh_object->mesh, obj_file);
+	if (return_code < 0)
+		return (-1);
+	else if (return_code == 0)
+		return (0);
```
Comment (V-Fries) [inline src/object/mesh/create.c:38]: Please add a define here before merging

**Review by V-Fries — APPROVED** [review]

**Pull request merged**